plugins/modules/cargo.py

```diff
@@ -187,43 +576,92 @@ def main():
         state=dict(default="present", choices=["present", "absent", "latest"]),
         version=dict(default=None, type="str"),
         locked=dict(default=False, type="bool"),
+        features=dict(default=[], type="list", elements="str"),
+        all_features=dict(default=False, type="bool"),
+        force=dict(default=False, type="bool"),
+        directory=dict(default=None, type="path"),
+        git=dict(default=None, type="str"),
+        branch=dict(default=None, type="str"),
+        tag=dict(default=None, type="str"),
+        rev=dict(default=None, type="str"),
+        registry=dict(default=None, type="str"),
+        debug=dict(default=False, type="bool"),
     )
     module = AnsibleModule(argument_spec=arg_spec, supports_check_mode=True)
 
     name = module.params["name"]
     state = module.params["state"]
     version = module.params["version"]
+    force = module.params["force"]
+    features = module.params["features"]
+    all_features = module.params["all_features"]
+    directory = module.params["directory"]
+    registry = module.params["registry"]
+    git = module.params["git"]
+    branch = module.params["branch"]
+    tag = module.params["tag"]
+    rev = module.params["rev"]
 
     if not name:
         module.fail_json(msg="Package name must be specified")
 
+    if len(name) > 1 and features:
+        module.fail_json(
+            msg="Cannot determine which features are for what package for multiple packages"
+        )
+
+    if len(name) > 1 and version:
+        module.fail_json(msg='Instead of version, use "crate@version"...')
+
+    if version and (directory or git or branch or tag or rev):
+        module.fail_json(msg="version option only works with a registry")
```

Comment:
Same here.

Reply:
Again, `directory`, `git`, etc. are recent additions, it will not break previously written Ansible playbooks.

Reply:
Yes, but you shouldn't check this manually, but use `AnsibleModule`'s facilities for it. `version` is mutually exclusive with `directory`, `git`, `branch`, `tag`, and `rev`.